backend_python/backend/game.py

```diff
@@ -0,0 +1,184 @@
+            on_exit=self.execute_vote))
+
+        states.append(State('night_results',
+            on_enter=self.stamp_state))
+            
+        states.append(State('game_ended', 
+            on_enter=[self.stamp_state, self.set_winner]))
+                
+        return states
+
+
+    def generate_transitions(self):
+        transitions = []
+        transitions.append({
+                'trigger': 't_start_game',
+                'source': 'waiting_for_players',
+                'dest': 'day_voting',
+                'conditions': self.town.check_ready_to_start,
+            })
+        transitions.append({
+                'trigger': 't_execute_vote',
+                'source': 'day_voting',
+                'dest': 'day_results',
+                'conditions': None,
+                'after': self.end_game_maybe,
+            })
+        transitions.append({
+                'trigger': 't_progress',
+                'source': 'day_results',
+                'dest': 'night_voting',
+                'conditions': None,
+                'before': self.progress,
+            })
+        transitions.append({
+                'trigger': 't_progress',
+                'source': 'night_voting',
+                'dest': 'night_results',
+                'conditions': self.can_game_continue,
+                'after': self.end_game_maybe, 
+            })
+        transitions.append({
+                'trigger': 't_progress',
+                'source': 'night_results',
+                'dest': 'day_voting',
+                'conditions': None,
+                'before': self.progress,
+            })
+        transitions.append({
+                'trigger': 't_end_game',
+                'source': ['day_results', 'night_results'],
+                'dest': 'game_ended',
+                'before': self.end_game
+            })
+        
+        return transitions
+
+
+    def to_dict(self):
+        dictionary = self.town.to_dict()
+        self.status['id'] = self.state
+        dictionary['state'] = self.status
+        dictionary['version'] = self.version
+
+        return dictionary
+
+    def jversonify(self):
+        self.version = datetime.now().timestamp()
+
+        return(jsonify(self.to_dict()))
+
+
+    def start_game(self, *args):
+        self.t_start_game()
+        self.town.assign_characters(*args)
+
+
+    def execute_vote(self, *args, **kwargs):
+        self.status['killed_player'] = self.town.execute_vote()
+
+    def progress(self, *args, **kwargs):
+        self.town.clear_vote_pool()
+    
+
+    def end_game(self, *args, **kwargs):
+        pass    
+        
+
+    def vote(self, voterName, voteeName):
+        if self.state not in Game.voting_states:
+            raise NotInVotingStateException
+        self.town.vote(voterName, voteeName)
+        
+
+    def is_state_outdated(self):
+        return self.status['started_at'] + Game.state_live_time < self.version
```

Comment:
there is a common acronym for that: **TTL** (Time To Live), i would rename `state_live_time` to `state_ttl`